In [3]:
from sklearn import datasets
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Perceptron
from sklearn.cross_validation import train_test_split
from sklearn.metrics import accuracy_score
import numpy as np
import pandas as pd

In [5]:
iris = datasets.load_iris()

In [6]:
X = iris.data
y = iris.target

In [8]:
y[:5]

array([0, 0, 0, 0, 0])

In [9]:
X[:5]

array([[ 5.1,  3.5,  1.4,  0.2],
       [ 4.9,  3. ,  1.4,  0.2],
       [ 4.7,  3.2,  1.3,  0.2],
       [ 4.6,  3.1,  1.5,  0.2],
       [ 5. ,  3.6,  1.4,  0.2]])

In [10]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size= 0.3)

In [11]:
sc= StandardScaler()
sc.fit(X_train)

StandardScaler(copy=True, with_mean=True, with_std=True)

In [12]:
X_train_std = sc.transform(X_train)

X_test_std = sc.transform(X_test)

In [13]:
perc = Perceptron(n_iter=40, eta0= 0.1,random_state=0)
perc.fit(X_train_std, y_train)

Perceptron(alpha=0.0001, class_weight=None, eta0=0.1, fit_intercept=True,
      n_iter=40, n_jobs=1, penalty=None, random_state=0, shuffle=True,
      verbose=0, warm_start=False)

In [14]:
Perceptron(alpha=0.0001, class_weight=None, eta0=0.1, fit_intercept=True,
      n_iter=40, n_jobs=1, penalty=None, random_state=0, shuffle=True,
      verbose=0, warm_start=False)

Perceptron(alpha=0.0001, class_weight=None, eta0=0.1, fit_intercept=True,
      n_iter=40, n_jobs=1, penalty=None, random_state=0, shuffle=True,
      verbose=0, warm_start=False)

In [16]:
y_pred = perc.predict(X_test_std)

In [17]:
y_pred

array([1, 0, 0, 2, 1, 0, 1, 0, 2, 1, 1, 0, 2, 2, 1, 0, 0, 1, 0, 1, 1, 0, 2,
       1, 1, 0, 0, 0, 1, 2, 0, 2, 0, 1, 0, 0, 2, 2, 2, 0, 2, 1, 1, 0, 0])

In [18]:
y_test

array([1, 0, 0, 2, 1, 0, 1, 0, 2, 1, 1, 0, 2, 2, 1, 0, 0, 2, 0, 1, 1, 0, 2,
       1, 1, 0, 0, 0, 1, 2, 0, 2, 0, 1, 0, 0, 2, 2, 2, 0, 2, 1, 1, 0, 0])

In [19]:
print('Accuracy: %2f' % accuracy_score(y_test, y_pred))

Accuracy: 0.977778


# Conclusion

We are able to implement single layer perceptron on iris dataset with an accuracy of 97.78%
